processing/src/main/java/org/apache/druid/segment/IndexMergerV9.java

```diff
@@ -332,6 +348,174 @@ private void makeMetadataBinary(
     }
   }
 
+  private void makeProjections(
```

Comment:
Looks like a lot of copied and tweaked stuff from elsewhere. Is it possible to refactor things so the copying isn't as needed? Maintenance of the code is going to be difficult since various changes would need to be made in both places.

Reply:
yea, still working on this. i started from a place of nearly total isolation to minimize risk, but have been chipping away at consolidating stuff.

Conceptually, i would like this to be basically the same as the refactor i would like to do eventually to incremental index, where basically the idea is "everything is a projection", the base table just happens to be a bit of a special one.